openarc/openarc-ar.c

```diff
@@ -501,6 +501,9 @@ ares_parse(u_char *hdr, struct authres *ar)
 				if (n > 0)
 					n--;
 
+				prevstate = state;
+				state = 2;
```

Comment:
I have a slight preference for your second idea, but anything that ensures that "none" is the last thing is fine.